src/main/java/Duke.java

```diff
@@ -1,10 +1,232 @@
+import java.io.IOException;
+import java.io.FileNotFoundException;
+
 public class Duke {
+
+    public static final String STRAIGHT_LINE = "-------------------------------------------";
+    public static final String FILE_PATH = "duke.txt";
+    public static ArrayList<Task> tasks = new ArrayList<>();
+
+    //write to the file
+    public static void saveToFile(String FILE_PATH){
+        try{
+            File f = new File(FILE_PATH);
+            FileWriter fw = new FileWriter(FILE_PATH);
+            for (Task task: tasks){
+                fw.write(task.saveToFile());
+            }
+            fw.close();
+        } catch (IOException e){
+            System.out.println("Something went wrong: " + e.getMessage());
+        }
+    }
+
+    //load from file
+    public static void loadFromFile(String FILE_PATH){
+        try{
+            File f = new File(FILE_PATH);
+            if (!f.exists()){
+                f.createNewFile();
+            }
+            Scanner s = new Scanner(f);
+            Task task;
+            while(s.hasNext()){
+                String[] readTask = s.nextLine().split("\\|");
+                switch (readTask[0]) {
+                case ("T"):
+                    task = new ToDo(readTask[2]);
+                    break;
+                case ("D"):
+                    task = new Deadline(readTask[2], readTask[3]);
+                    break;
+                case ("E"):
+                    task = new Event(readTask[2], readTask[3]);
+                    break;
+                default:
+                    task = new Task(readTask[2]);
+                }
+
+                if (readTask[1].equals("true")){
+                    task.setIsDone(true);
+                }
+                tasks.add(task);
+            }
+        } catch (FileNotFoundException e){
+            System.out.println("File not found");
+        } catch (IOException e){
+            e.printStackTrace();
+            System.out.println("File creation failed");
+        }
+    }
+
+    //print the horizontal line
+    public static void printHorizontalLine() {
+        System.out.println(STRAIGHT_LINE);
+    }
+
+    //print a welcome message
+    public static void printWelcomeMessage() {
+        printHorizontalLine();
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?\n");
+        printHorizontalLine();
+    }
+
+    //print a goodbye message
+    public static void printByeMessage() {
+        printHorizontalLine();
+        System.out.println("Bye. Hope to see you again soon!\n");
+        printHorizontalLine();
+    }
+
+    //echo the message
+    public static void echoMessage(Task message, int index) {
+        printHorizontalLine();
+        System.out.println("Got it. I've added this task: ");
+        System.out.println(message.toString());
+        System.out.println("Now you have " + index + " tasks in the list");
+        printHorizontalLine();
+    }
```

Comment:
maybe one method like this printMsg(msg){ printHorizontalLine(); system.out.println(msg);printHorizontalLine();}?Then for each echo/print msg you call printMsg(yourspecificmsg). This should alleviate your the redundancies.